modules/ui/source_subfield.js

```diff
@@ -0,0 +1,173 @@
+import {
+    select as d3_select
+} from 'd3-selection';
+
+import { t } from '../core/localizer';
+import { svgIcon } from '../svg/icon';
+import { uiTooltip } from './tooltip';
+import { utilGetSetValue, utilUniqueDomId } from '../util';
+
+export function uiSourceSubfield(context, field, tags, dispatch) {
+
+    var sourceSubfield = {};
+
+    let sourceInput = d3_select(null);
+    let sourceKey = field.key + ':source';
+    let _sourceValue = tags[sourceKey];
+
+    function renderSourceInput(selection) {
```

Comment:
If this code was adapted from another input function, add a comment mentioning the original function so that we can more easily keep them synchronized in the future.

(For context, we’re looking at adding support for Rapid in the future: OpenHistoricalMap/issues#684. Rapid has refactored the field UI code to streamline some of the boilerplate code you’ve had to deal with.)